imageio/plugins/tifffile.py

```diff
@@ -0,0 +1,153 @@
+    resolution : (float, float) or ((int, int), (int, int))
+        X and Y resolution in dots per inch as float or rational numbers.
+    description : str
+        The subject of the image. Saved with the first page only.
+    compress : int
+        Values from 0 to 9 controlling the level of zlib compression.
+        If 0, data are written uncompressed (default).
+    volume : bool
+        If True, volume data are stored in one tile (if applicable) using
+        the SGI image_depth and tile_depth tags.
+        Image width and depth must be multiple of 16.
+        Few software can read this format, e.g. MeVisLab.
+    writeshape : bool
+        If True, write the data shape to the image_description tag
+        if necessary and no other description is given.
+    extratags: sequence of tuples
+        Additional tags as [(code, dtype, count, value, writeonce)].
+
+        code : int
+            The TIFF tag Id.
+        dtype : str
+            Data type of items in 'value' in Python struct format.
+            One of B, s, H, I, 2I, b, h, i, f, d, Q, or q.
+        count : int
+            Number of data values. Not used for string values.
+        value : sequence
+            'Count' values compatible with 'dtype'.
+        writeonce : bool
+            If True, the tag is written to the first page only.
+    """
+
+    def _can_read(self, request):
+        if request.filename.lower().endswith(TIFF_FORMATS):
+            return True  # We support any kind of image data
+        else:
+            return False
+
+    def _can_write(self, request):
+        if request.filename.lower().endswith(TIFF_FORMATS):
+            return True  # We support any kind of image data
+        else:
+            return False
+
+    # -- reader
+
+    class Reader(Format.Reader):
+
+        def _open(self):
+            if not tifffile:
+                load_tifffile()
+            self._tf = tifffile.TiffFile(self.request.get_local_filename())
+
+        def _close(self):
+            self._tf.close()
+
+        def _get_length(self):
+            return len(self._tf)
+
+        def _get_data(self, index):
+            # Get data
+            if index < 0 or index >= len(self._tf):
+                raise IndexError(
+                    'Index out of range while reading from tiff file')
+            im = self._tf[index].asarray()
+            # Return array and empty meta data
+            return im, {}
+
+        def _get_meta_data(self, index):
+            raise RuntimeError('The tiff format does not support meta data.')
+
+    # -- writer
+
+    class Writer(Format.Writer):
+
+        def _open(self, bigtiff=None, byteorder=None, software=None,
+                  **kwargs):
+            self._parameters = kwargs
+            if not tifffile:
+                load_tifffile()
+            self._tf = tifffile.TiffWriter(self.request.filename,
+                                           bigtiff, byteorder, software)
```

Comment:
This should be `request.get_local_filename()` or `request.get_file()`. `request.filename` can also be an http address or zip file.